src/core/flow.js

```diff
@@ -631,6 +676,63 @@ export function Flow({ moves, phases, endIf, turn, events, plugins }) {
     return { ...state, G, ctx, deltalog, _undo: [], _redo: [] };
   }
 
+  function EndStage(state, { arg, next, automatic, playerID }) {
+    playerID = playerID || state.ctx.currentPlayer;
+
+    let { ctx } = state;
+    let { activePlayers, _activePlayersMoveLimit } = ctx;
+
+    if (next) {
+      next.push({ fn: UpdateStage, arg, playerID });
```

Comment:
Don't add UpdateStage to `next` if `arg` is undefined (as opposed to adding it and then having `UpdateStage` being a no-op).